### PR TITLE
Added: Filters

### PR DESCRIPTION
**Pull request by KaustubhM1997**
Features:

1) Filter by price (Low to High and High to Low)
2) Filter By Category (Men, Women, Kids, Best-sellers)
3) Filter by Availability and Fast Delivery
4) Price range slider
5) Filter by Ratings
6) Clear (takes back to initial values)

Live preview - https://endearing-empanada-84db11.netlify.app/

### CODE REVIEW

File: src/backend/db/products.js [hunk 1]
```diff
@@ -6,25 +6,156 @@ import { v4 as uuid } from "uuid";
...
+
+    // price range: 100 to 2000, step of 500
+    _id: uuid(),
+    title: "Casual Blue Shirt", 
+    price: 599,
+    rating: "3.7",
+    categoryName: "Men",
+    inStock: true,
+    fastDelivery: true,
+    categoryImg: "https://media.istockphoto.com/photos/latin-man-standing-picture-id825083570?b=1&k=20&m=825083570&s=170667a&w=0&h=Pt9O2-RyYt84bumpiuy3KDUvaDIm--UYutUUnwBKVZM="
```
Comment (mimansha-swarup) [inline src/backend/db/products.js]: technically its product image not category image

File: src/components/Aside/aside.jsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            checked = {state.fastDelivery}
+             />
+            <p className="checkbox-notify">Fast Delivery</p>
+          </div>
+        </div>
+
+        <div className="filter-category flex-column">
+          <h3 className="filter-headings">Category</h3>
+          <div className="input-checkbox-container">
+            {/* {console.log(state.category)} */}
```
Comment (mimansha-swarup) [inline src/components/Aside/aside.jsx]: remove console.log

File: src/components/Product-card/product-card.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+          <div className="rating-badge">
+            {rating}
+            <span className="star-icon">
+              <i class="fa-solid fa-star"></i>
+            </span>
+          </div>
+
+          {/* <p className="card-subtext">
+              A bit more detail about the product
```
Comment (mimansha-swarup) [inline src/components/Product-card/product-card.jsx:26]: remove unwanted comments
keep only meaningful comments

File: src/reducers/productReducer.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    case "CATEGORY":
+      const categoryExist = state.category.includes(action.payload);
+
+      return categoryExist
+        ? {
+            ...state,
+            category: state.category.filter((item) => item !== action.payload),
+          }
+        : { ...state, category: [...state.category, action.payload] };
+    // in order to select multiple filters we just take an array and add any new filter to the ones already present by spreading them. We also put a condition to uncheck the ones that are selected
```
Comment (mimansha-swarup) [inline src/reducers/productReducer.jsx:24]: great practice of writing meaningful comments

**Review by mimansha-swarup — APPROVED** [review]
all looks good to me
one more point link your label to checkboxes cause its not working for checkboxes its working alright for radio

File: src/components/Product-card/product-card.jsx (hunk 3 above)
Comment (KaustubhM1997) [reply src/components/Product-card/product-card.jsx:26]: Thank you, I might use this later so I've kept it.

File: src/components/Aside/aside.jsx (hunk 2 above)
Comment (KaustubhM1997) [reply src/components/Aside/aside.jsx]: Done, thank you

File: src/backend/db/products.js (hunk 1 above)
Comment (KaustubhM1997) [reply src/backend/db/products.js]: yeah, makes sense. Thanks for pointing it out.

File: src/reducers/productReducer.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  switch (action.type) {
+    case "SORT":
+      return { ...state, sortItemsBy: action.payload };
+    case "FILTER_BY_RATINGS":
+      return { ...state, rateBy: action.payload };
+    case "IN_STOCK":
+      return { ...state, inStock: action.payload };
+    case "FAST_DELIVERY":
+      return { ...state, fastDelivery: action.payload };
+    case "CATEGORY":
```
Comment (BelideNikhil) [inline src/reducers/productReducer.jsx:15]: As suggested by TA's, you can save action types as variables

**Review by BelideNikhil — APPROVED** [review]
Filtering works nicely
Clean code
Good to see that you've done barrel export everywhere

**Pull request merged**